### DISCUSSION TITLE
Custom breakpoint TypeScript issue

### DISCUSSION THREAD

**Discussion by Mahmoud-Azamtta** [chakra-ui/chakra-ui#9083] [Q&A] [answered]
This is my first time using Chakra-UI
I installed it following the steps in the documentation, and wanted to make a custom theme. So I started by create a custom breakpoint, I copied the code from <a target="_blank" href="https://www.chakra-ui.com/docs/theming/customization/breakpoints">here</a>
here is the code for the `theme.ts`
```ts
import { defineConfig, defaultConfig, createSystem } from "@chakra-ui/react";

const config = defineConfig({
  theme: {
    breakpoints: {
      desktop: "1200px",
    },
  },
});

export const system = createSystem(defaultConfig, config);
```

I saved, ran `chakra typegen theme.ts`, and finally I restarted my LSP. **The breakpoint works fine when I run the app**.
But still I keep getting a type error here on the desktop breakpoint:
```tsx
import { Button } from "@chakra-ui/react";
import React from "react";

const Demo: React.FC = () => {
  return (
    <Button
      bg={{
        desktop: "cyan.400",
      }}
      padding={"8"}
      _hover={{ bg: { "2xl": "cyan.700" } }}
    >
      Click me
    </Button>
  );
};

export default Demo;
```

I don't know what is wrong, cuz as I said this is my first time using it and I might be doing something wrong. 

I use the following: 
- react 18
- next 14.2.16
- chakra 3.0.2
- typescript 5

Any suggestions on what would be the issue.
Thanks in advance

**Comment by ericksonholguin** [comment]
@Mahmoud-Azamtta Have you received an update about that? I have the same problem and I don't know what to do

**Reply by Mahmoud-Azamtta** [reply]
I haven't actually
I decided to stick with the built in breakpoints, although it would have been nice if had the ability to create my own without keep getting the type check errors.🤷‍♂️

**Reply by ericksonholguin** [reply]
@Mahmoud-Azamtta I opened a issue for this situation https://github.com/chakra-ui/chakra-ui/issues/9168, so, you can take a look

**Comment by segunadebayo** [comment, marked as answer]
The breakpoint works at runtime. TypeScript does not see it until the typegen file is in your program.

Point `tsconfig` at the generated file (or the `include` path the CLI prints), then restart the TS server.
